src/main/scala/com/nec/native/CppTranspiler.scala

```diff
@@ -300,57 +342,96 @@ object CppTranspiler {
           // Fetch values
           signature.inputs.map { in => s"${in.name}_val = ${in.name}[0]->data[i];" },
           "",
-          // Perform map operation and set validity
+          // Perform the map operation and set validity
           mapOperation,
           signature.outputs.map { d => s"${d.name}[0]->set_validity(i, 1);" }
         )
       }
     )
   }
 
-  def evalMapFunc(func: Function): String = {
-    val signature = func.veMapSignature
-    val codelines = func.body match {
+  private def mapOperation(block: Block, signature: VeSignature): CodeLines = {
+    /*
+      The last line of the code block is the only line that is transpiled
+      differently because its return value(s) need to be written into the output
+      vecs.
+    */
+    val lastline = block.children.last match {
       case ident @ Ident(name) =>
-        CodeLines.from(s"${signature.outputs.head.name}[0] = ${evalIdent(ident).replace("_val", "")}[0]->clone();")
+        CodeLines.from(s"${signature.outputs.head.name}[0]->data[i] = ${evalIdent(ident)};")
 
-      case apply @ Apply(_, _) =>
-        val mapOperation = CodeLines.from {
+      case apply: Apply =>
+        CodeLines.from {
           (apply.fun, apply.args) match {
             case (TypeApply(Select(Ident(ident), TermName("apply")), _), args) if ident.toString.startsWith("Tuple") =>
               args.zip(signature.outputs).map { case (tupleArg, out) =>
-                s"${out.name}[0]->data[i] = ${evalArg(tupleArg)};"
+                s"${out.name}[0]->data[i] = ${evalQual(tupleArg)};"
               }
 
             case _ =>
               List(s"out_0[0]->data[i] = ${evalApply(apply)};")
           }
         }
-        mapCode(signature, mapOperation)
 
-      case select @ Select(_, _) =>
-        val mapOperation = CodeLines.from(s"out_0[0]->data[i] = ${evalSelect(select)};")
-        mapCode(signature, mapOperation)
+      case ifblock: If =>
+        CodeLines.from(s"out_0[0]->data[i] = ${evalIf(ifblock)};")
+
+      case select: Select =>
+        CodeLines.from(s"out_0[0]->data[i] = ${evalSelect(select)};")
```

Comment:
How are tuple outputs handled in those cases?

Reply:
Tuple outputs should be handled in the `case apply: Apply` case roughly 20 lines above where this comment is located (second case of `mapOperation`), if I understood your question correctly.